doc/design/volumes/nfsvolumestores.md

```diff
@@ -70,33 +70,53 @@ The advantage to using the interface is the storage layer maintains consistency
  48 }
 ```
 
-When we create the NFS `VolumeStore`, we'll store the NFS target parameters (`host` + `path`) in the implementation's struct.  This is the only information we'll need to mount the NFS target on the container.
+When we create the NFS `VolumeStore`, we'll store the NFS target parameters (`host` + `path`) in the implementation's struct.  This is the only information we'll need to mount the NFS target on the container. NFS based volume stores will work for the default volume store as well.
 
 ```
-type NFSv3VolumeStore struct {
- target *url.URL
+
+// VolumeStore this is nfs related volume store definition
+type VolumeStore struct {
+	// volume store name
+	Name string
+
+	// Service is the interface to the nfs target.
+	Service MountServer
+
+	// Service selflink to volume store.
+	SelfLink *url.URL
 }
+
+
 ```
 
-_The implementation is still being worked on.  The open question is whether the VCH appliance will mount the target to manipulate the NFS target, or use an NFS client implementation in userspace instead.  I'd much (*MUCH*) rather do the latter.  The `linux` VFS implementation throws `sync` errors when mounts are unavailable.  And we don't want to bring down the appliance because of a network hiccup.  NFS is a simple protocol and there is a public pkg which implements most of it.  Adding the few primitives we need shouldn't be that difficult, but more evaluation of the work required is needed_
+
+On creation the volume store target path is mount to the VCH and the volumes directory and metadata direectories are made under that target path. The client used is one that sits in the user space of the portlayer, this is from the vendored clientin our vendored code [For Reference](github.com/fdawg4l/go-nfs-client/nfs). This avoids the issue of The `linux` VFS implementation throwing `sync` errors when mounts are unavailable.
+
+As for Containers the volume will be mounted during tether boot time, this is using the man 2 version of the nfs mount call found in the golang syscall package. Currently, the mount options are not configurable, there are plans to address this going forward as an improvement to the nfs volume implementation. These will be configurable by the VI admin at volume store creation time. They will be configured as query paramters for the url supplied as an nfs target. 
```

Comment:
I might change this to something like `In a container, the volume will be mounted during tether boot time, using the man (2) version...`